### PR TITLE
removed all react import in components

### LINKED ISSUE #1222: Clean up usage of import React in libraries
We have a few conflicting linting rules with `react` not having a default export which forces us to ignore linting to check if modules have no default export. This can cause unwanted errors in the future if we introduce new libraries or existing ( popperjs, downshift, ramda etc) change their apis.

Either we change the remaining imports the cause errors to `import * as React from 'react'` or just remove `react` imports all together as they are not needed in React 17 anymore.

Notes: 
* https://javascript.plainenglish.io/react-17-new-jsx-transform-c6f983f3464d

### PR DESCRIPTION
**Pull request by Serjeel-Ranjan-911**
I have removed react import from all files possible.
```
'react/react-in-jsx-scope': 'off',
```
I didn't add the above line but it's still passing all the tests. 
Please let me know if I have missed something. fixes #1222

### CODE REVIEW

**Review by pomfrida — CHANGES_REQUESTED** [review]
Hi and thanks for your contribution! 👍  Please note your changes are not on our actual components (just Storefront components). All the EDS components are located in `design-system/libraries/core-react/src`

**Comment by Serjeel-Ranjan-911** [comment]
> Hi and thanks for your contribution! 👍 Please note your changes are not on our actual components (just Storefront components). All the EDS components are located in `design-system/libraries/core-react/src`

ohh ok. I thought I don't have to touch libraries. I will fix in next commit.

**Comment by Serjeel-Ranjan-911** [comment]
@pomfrida also what should I do with this import?
```
import * as React from 'react'
```

**Comment by mimarz** [comment]
> I have removed react import from all files possible.
> 
> ```
> 'react/react-in-jsx-scope': 'off',
> ```
> 
> I didn't add the above line but it's still passing all the tests.
> Please let me know if I have missed something. fixes #1222

The linting might fail, but we'll see! 

Thanks for taking the time to help us clean up! 😊

**Comment by Serjeel-Ranjan-911** [comment]
> @pomfrida also what should I do with this import?
> 
> ```
> import * as React from 'react'
> ```

should I let it be there?

**Comment by pomfrida** [comment]
> @pomfrida also what should I do with this import?
> 
> ```
> import * as React from 'react'
> ```

Please remove this too 👍

**Comment by Serjeel-Ranjan-911** [comment]
> > @pomfrida also what should I do with this import?
> > ```
> > import * as React from 'react'
> > ```
> 
> Please remove this too 👍

The typescript files are not allowing it's removal.  maybe refer [here](https://stackoverflow.com/questions/55285737/import-as-react-from-react-vs-import-react-from-react). 
image
there are some functions using that import.

**Comment by mimarz** [comment]
> > > @pomfrida also what should I do with this import?
> > > ```
> > > import * as React from 'react'
> > > ```
> > 
> > 
> > Please remove this too 👍
> 
> The typescript files are not allowing it's removal. maybe refer [here](https://stackoverflow.com/questions/55285737/import-as-react-from-react-vs-import-react-from-react).
> image
> there are some functions using that import.


What if you import them like the util functions?  

I think this should work:
```typescript
import type { KeyboardEvent } from 'react'
```

**Comment by Serjeel-Ranjan-911** [comment]
> > > > @pomfrida also what should I do with this import?
> > > > ```
> > > > import * as React from 'react'
> > > > ```
> > > 
> > > 
> > > Please remove this too 👍
> > 
> > 
> > The typescript files are not allowing it's removal. maybe refer [here](https://stackoverflow.com/questions/55285737/import-as-react-from-react-vs-import-react-from-react).
> > image
> > there are some functions using that import.
> 
> What if you import them like the util functions?
> 
> I think this should work:
> 
> ```ts
> import type { KeyboardEvent } from 'react'
> ```

Even components are having problems
```
'React' refers to a UMD global, but the current file is a module. Consider adding an import instead.
```
I receive this error for components. 

image

**Comment by mimarz** [comment]
Ah, this is a Typescript compiler error, looks like we need to [change the compiler-target](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1-beta/#jsx-factories) for `jsx` in our `tsconfig`. 

You can change this in our base.tsconfig [here](https://github.com/equinor/design-system/blob/b2045bb8e1b0149be02211f74dd63e284ab9d442/tsconfig.base.json#L16).

setting it from `react` to `react-jsx` seemed to do the trick

**Comment by Serjeel-Ranjan-911** [comment]
@mimarz help needed.

**Comment by mimarz** [comment]
> @mimarz help needed.

Do you have linting enabled locally? :) You can see the error [here](https://github.com/equinor/design-system/pull/1251/checks?check_run_id=2394147777#step:3:395) nonetheless. Just a missing newline :)

**Comment by Serjeel-Ranjan-911** [comment]
> > @mimarz help needed.
> 
> Do you have linting enabled locally? :) You can see the error [here](https://github.com/equinor/design-system/pull/1251/checks?check_run_id=2394147777#step:3:395) nonetheless. Just a missing newline :)

sorry I don't have linter enabled. The error is so big I don't know which line needs a newline. I have not worked with linter before.

**Comment by mimarz** [comment]
> > > @mimarz help needed.
> > 
> > 
> > Do you have linting enabled locally? :) You can see the error [here](https://github.com/equinor/design-system/pull/1251/checks?check_run_id=2394147777#step:3:395) nonetheless. Just a missing newline :)
> 
> sorry I don't have linter enabled. The error is so big I don't know which line needs a newline. I have not worked with linter before.

There is only one error, and the link above highlights it, its this one if it wasn't clear: 
```
/home/runner/work/design-system/design-system/libraries/core-react/src/components/Table/Inner.context.tsx
  1:1  error  Expected 1 empty line after import statement not followed by another import  import/newline-after-import
```

**Comment by Serjeel-Ranjan-911** [comment]
@mimarz  finally all tests passed. I should have enabled linter in my pc.

File: libraries/core-react/src/components/Accordion/AccordionItem.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import * as React from 'react'
+import { Children as ReactChildren, cloneElement } from 'react'
```
Comment (mimarz) [inline libraries/core-react/src/components/Accordion/AccordionItem.tsx:1]: Looks like you ended up with double import statements from `react` (line 1 and line 9) here, so suggest changing it to just one :)

**Review by mimarz — CHANGES_REQUESTED** [review]
Good piece of work here! 👍
Just noticed some small irregularities that probably snuck in while refactoring so many lines! 😅

**Review by mimarz — APPROVED** [review]
Thank you very much for contributing @Serjeel-Ranjan-911  👍 

Appreciate the help!

**Comment by Serjeel-Ranjan-911** [comment]
> Thank you very much for contributing @Serjeel-Ranjan-911  👍 
> 
> Appreciate the help!

@mimarz your welcome. I never contributed to such a big repo before. I am trying to learn open source contribution. Also if anyone interested can exchange follow with me :)

Typescript was really useful for immediately finding where the code is breaking, although I never coded on it maybe I will consider using it.

**Review by pomfrida — APPROVED** [review]
LGTM 👍

**Comment by mimarz** [comment]
> > Thank you very much for contributing @Serjeel-Ranjan-911  👍
> > Appreciate the help!
> 
> @mimarz your welcome. I never contributed to such a big repo before. I am trying to learn open source contribution. Also if anyone interested can exchange follow with me :)
> 
> Typescript was really useful for immediately finding where the code is breaking, although I never coded on it maybe I will consider using it.

Always good to contribute to open-source projects :) Might I ask where you heard about this project? Always interesting to hear :) @Serjeel-Ranjan-911

**Comment by Serjeel-Ranjan-911** [comment]
@mimarz  truly speaking I just roam around in issues section and work on issues which I can.

**Pull request merged**